eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -1018,6 +1076,427 @@ def test_update_valid_enrollment_using_force_flag(self, param: str) -> None:
         self.assertTrue(enrollment_response_data["is_active"])
 
 
+@ddt.ddt
+class TestPreEnrollmentAPIIntegration(
+    BaseAPIIntegrationTest,
+    UsersAPIRequestMixin,
+    PreEnrollmentAPIRequestMixin,
+):
+    """Integration test suite for the Pre-Enrollment API"""
+
+    def setUp(self) -> None:
+        """Set up the test suite"""
+        super().setUp()
+        self.user = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, self.user)
+
+    @ddt.data(
+        {"auto_enroll": True},
+        {"auto_enroll": False},
+        {"email": "user-not-found1@mail.com", "auto_enroll": True},
+        {"email": "user-not-found2@mail.com", "auto_enroll": False},
+    )
+    def test_create_pre_enrollment_success(self, data: dict) -> None:
+        """
+        Test creating an pre-enrollment with a valid course and a user in a tenant.
+
+        The pre-enrollment is created even if the user does not exist in the tenant.
+
+        Open edX definitions tested:
+        - `validate_org`
+        - `get_valid_course_key`
+        - `CourseEnrollmentAllowed`
+        - `get_courseware_courses`
+
+        Expected result:
+        - The status code is 200.
+        - The response contains the pre-enrollment data.
+        - The pre-enrollment is created in the tenant.
+        """
+        pre_enrollment_data = {
```

Comment:
Can we ensure that all the tests follow the AAA pattern and leave a blank line separating each section? In this case, it would look something like:
```
pre_enrollment_data = {
...
}

response = 
response_data = 

pre_enrollment_response = ... # Not entirely sure about this being here, but since this is another action that is later asserted, looks like it should be in the Act section as well

self.assertEqual
...
```

Reply:
Done: https://github.com/eduNEXT/eox-core/commit/7f13278e8e643fc2abdcdae60cbb0c4605bfe51e